### PR TITLE
Add new exercise linked-list

### PR DESCRIPTION
**Pull request by matthewmorgan**
As described in #122

### CODE REVIEW

File: linked-list/linked-list.spec.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import LinkedList from './linked-list';
+
+describe('LinkedList', () => {
+  it('push/pop', () => {
```
Comment (rchavarria) [inline linked-list/linked-list.spec.js]: Could a better description be: `add/extract elements to the end of the list with push/pop`?

File: linked-list/linked-list.spec.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    const list = new LinkedList();
+    list.push(10);
+    list.push(20);
+    expect(list.pop()).toBe(20);
+    expect(list.pop()).toBe(10);
+  });
+  xit('push/shift', () => {
```
Comment (rchavarria) [inline linked-list/linked-list.spec.js]: Could a better description be: `extract elements from the beginning of the list with shift`?

File: linked-list/linked-list.spec.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    const list = new LinkedList();
+    list.push(10);
+    list.push(20);
+    expect(list.shift()).toBe(10);
+    expect(list.shift()).toBe(20);
+  });
+  xit('unshift/shift', () => {
```
Comment (rchavarria) [inline linked-list/linked-list.spec.js]: Could a better description be: `add/extract elements from the beginning of the list with unshift/shift`?

File: linked-list/linked-list.spec.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    list.push(50);
+    expect(list.shift()).toBe(40);
+    expect(list.pop()).toBe(50);
+    expect(list.shift()).toBe(30);
+  });
+  xit('pops undefined when there are no elements', () => {
+    const list = new LinkedList();
+    expect(list.pop()).toBe(undefined);
+  });
+  xit('can count xits elements', () => {
```
Comment (rchavarria) [inline linked-list/linked-list.spec.js]: Is `xits` a typo? Should it be `list`?

File: linked-list/linked-list.spec.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    const list = new LinkedList();
+    expect(list.count()).toBe(0);
+    list.push(10);
+    expect(list.count()).toBe(1);
+    list.push(20);
+    expect(list.count()).toBe(2);
+  });
+  xit('deletes the last element from the list', () => {
```
Comment (rchavarria) [inline linked-list/linked-list.spec.js]: The `delete` implementation deletes the element with the specified value, not the last element. The test description should reflect this.

File: linked-list/linked-list.spec.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    list.delete(20);
+    expect(list.count()).toBe(2);
+    expect(list.pop()).toBe(30);
+    expect(list.shift()).toBe(10);
+  });
+  xit('deletes the only element', function() {
+    const list = new LinkedList();
+    list.push(10);
+    list.delete(10);
+  });
```
Comment (rchavarria) [inline linked-list/linked-list.spec.js]: A expectation is missing here:

```
expect(list.count()).toBe(0);
```

File: linked-list/linked-list.spec.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    list.push(30);
+    list.delete(20);
+    expect(list.count()).toBe(2);
+    expect(list.pop()).toBe(30);
+    expect(list.shift()).toBe(10);
+  });
+  xit('deletes the only element', function() {
+    const list = new LinkedList();
+    list.push(10);
+    list.delete(10);
```
Comment (rchavarria) [inline linked-list/linked-list.spec.js]: A expectation is missing at the end of the test:

```
expect(list.count()).toBe(0);
```

File: linked-list/linked-list.spec.js (hunk 6 above)
Comment (rchavarria) [inline linked-list/linked-list.spec.js]: A new test could be added about `delete`:

```
xit('deletes does not modify the list if the element is not found', () => {
  const list = new LinkedList();
  list.push(10);
  list.delete(20);
  expect(list.count()).toBe(1);
});
```

File: linked-list/linked-list.spec.js [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+    const list = new LinkedList();
+    list.push(10);
+    list.push(20);
+    list.push(30);
+    list.delete(20);
+    expect(list.count()).toBe(2);
+    expect(list.pop()).toBe(30);
+    expect(list.shift()).toBe(10);
+  });
+  xit('deletes the only element', function() {
```
Comment (rchavarria) [inline linked-list/linked-list.spec.js]: An arrow function could be used here.

**Comment by rchavarria** [comment]
I commented on some lines. They could be suggestions for the xjavascript track. If so, I can take care of it. I'll wait for your opinion.

**Comment by matthewmorgan** [comment]
I think we can change this-- does it make sense to change the test as a separate PR?  Either way is fine with me.

File: linked-list/linked-list.spec.js (hunk 4 above)
Comment (kytrinyx) [inline linked-list/linked-list.spec.js]: A bit of a tangent, but my guess is that it was `its` and a search/replace on `it` with `xit` turned it into `xits`.

Comment (matthewmorgan) [inline linked-list/linked-list.spec.js]: I thought it was maybe Catalan...

**Comment by rchavarria** [comment]
I think it would be better changed and incorporated in this PR. The better the code, the better the PR :wink:

File: linked-list/linked-list.spec.js (hunk 4 above)
Comment (rchavarria) [inline linked-list/linked-list.spec.js]: It looks so obvious to me now that it was `its`, that I don't understand why I didn't see it before.

Could this test description be changed to `can count its elements`?

**Comment by matthewmorgan** [comment]
I will make the change.

**Comment by rchavarria** [comment]
Thanks for take care of them.

While merging the PR, I deleted a `console.log` statement.

**Comment by matthewmorgan** [comment]
Oops...thanks!

**Pull request merged**